k8s/k8s.go

```diff
@@ -388,6 +389,18 @@ func ipAddrsResult(ipAddrs string, conf utils.NetConf, args *skel.CmdArgs, logge
 			return nil, fmt.Errorf("Invalid IP format: %s", ip)
 		}
 
+		if ipAddr.To4() != nil {
+			if hasIPv4 {
+				return nil, fmt.Errorf("cannot have more than one IPv4 address for \"cni.projectcalico.org/ipAddrs\" annotation")
+			}
+			hasIPv4 = true
+		} else {
+			if hasIPv6 {
+				return nil, fmt.Errorf("cannot have more than one IPv6 address for \"cni.projectcalico.org/ipAddrs\" annotation")
+			}
+			hasIPv6 = true
+		}
```

Comment:
This will allocate at least one IP address before it fails - are we sure the tidy up will work correctly?

Should we check upfront that there is only a single IPv4 and single IPv6 address in the list *before* doing any assignment.